src/transaction/locator_sr.c

```diff
@@ -13937,6 +13941,166 @@ has_errors_filtered_for_insert (std::vector<int> error_filter_array)
 }
 // *INDENT-ON*
 
+
+static inline void
+locator_initialize_classname_entry (LOCATOR_CLASSNAME_ENTRY * entry, const char *classname, const OID * classoid)
+{
+  assert (classname != nullptr);
+
+  /* Memory for classname is allocated in atomic_replicator::update_classname_cache_for_ddl () */
+  entry->e_name = (char *) classname;
+  entry->e_tran_index = NULL_TRAN_INDEX;
+  entry->e_current.action = LC_CLASSNAME_EXIST;
+  COPY_OID (&entry->e_current.oid, classoid);
+  LSA_SET_NULL (&entry->e_current.savep_lsa);
+  entry->e_current.prev = nullptr;
+}
+
+/* locator_put_classname_entry () - Put a new entry in the locator_Mht_classnames, used in PTS replicator (atomic_replicator)
+ *
+ * return	     : error code
+ * thread_p (in)     : thread entry
+ * classname (in)    : classname to be inserted
+ * classoid (in)     : class oid of the classname
+ */
+int
+locator_put_classname_entry (THREAD_ENTRY * thread_p, const char *classname, const OID * classoid)
+{
+  assert (classname != nullptr);
+  assert (strlen (classname) < DB_MAX_IDENTIFIER_LENGTH);
+
+  LOCATOR_CLASSNAME_ENTRY *entry;
+
+  if (csect_enter (thread_p, CSECT_LOCATOR_SR_CLASSNAME_TABLE, INF_WAIT) != NO_ERROR)
+    {
+      assert (false);
+      return ER_FAILED;
+    }
+  // *INDENT-OFF*
+  scope_exit <std::function <void (void)>> classname_csect_exit ([&thread_p] ()
```

Comment:
Nit. I think it can be simplified with the _template_agrument_deduction_.
```suggestion
  auto classname_csect_exit = scope_exit {([&thread_p] ()
```

Reply:
Thanks!!